### PR TITLE
Add protected minio storage

### PR DESCRIPTION
**Pull request by jmsmkn**
This PR allows protected data to be stored in a minio instance rather than on disk. Users can access the data by requesting it from django, which then checks the permissions and returns an X-Accel-Redirect if the user is able to access the file and the file exists. NGINX then sets up an **internal** location block at `/$PROTECTED_S3_STORAGE_BUCKET_NAME/`.

### CODE REVIEW

File: app/grandchallenge/core/storage.py [hunk 1]
```diff
@@ -4,27 +4,43 @@
...
+    def __init__(self, *args, **kwargs):
+        super().__init__(
+            *args, config=settings.PRIVATE_S3_STORAGE_KWARGS, **kwargs
+        )
+
+
+@deconstructible
+class ProtectedS3Storage(S3Storage):
+    def __init__(self, *args, internal=False, **kwargs):
+        config = {**settings.PROTECTED_S3_STORAGE_KWARGS}
```
Comment (jmsmkn) [inline app/grandchallenge/core/storage.py]: @pkcakeout Should I use deepcopy here?

**Comment by jmsmkn** [comment]
- [ ] use deepcopy
- [ ] check that the buckets are different at runtime
- [ ] add a test for the accelerated redirect header

File: app/grandchallenge/core/storage.py (hunk 1 above)
Comment (pkcakeout) [reply app/grandchallenge/core/storage.py]: yes!

**Comment by pkcakeout** [comment]
Looks good as far as I understand it. When the changes from your checklist were applied, this is ready to be merged, I would say. One more time: This is a very neat and elegant solution using  the internal-redirect feature of NGINX.

**Pull request merged**